### DISCUSSION TITLE
Differentiating between patients that have Age = 89 and Age >89 when Deidentification for Patient is Enabled

### DISCUSSION THREAD

**Discussion by lawdank** [uwrit/leaf#602] [Q&A]
Hi Leaf Folks,

We have De-identification enabled for Patients.
When Querying Patients with Ages >=89 the Age is reflected as 89 for all.
<img width="52" alt="image" src="https://github.com/uwrit/leaf/assets/26530107/81f91e60-d900-4c82-9821-a70a1f206b2b">

We really need a way to differentiate between people who are actually 89 years old and those that are older than 89 years old.

Having Ages shown as 89 and >89 would be great.
Could you help us figure out a way to achieve this?

Thanks,
Sheersha

**Comment by ndobb** [comment]
Hi Sheersha,

Thanks for your patience in our reply. I think the challenge here is that we originally [defined the Age variable as an integer](https://github.com/uwrit/leaf/blob/master/src/server/Model/Cohort/PatientDemographic.cs#L28), whereas the ideal value shown to users would likely be something like `>89`, which would need to be a `string`.

The steps needed to do so would be (off the top of my head):

- Redefine the `PatientDemographic` class `Age` variable to be a string in both the [API](https://github.com/uwrit/leaf/blob/master/src/server/Model/Cohort/PatientDemographic.cs#L28) and [client](https://github.com/uwrit/leaf/blob/master/src/ui-client/src/models/patientList/DatasetDefinitionTemplate.ts#L22).
- Rewrite the [DemographicAggregator](https://github.com/uwrit/leaf/blob/master/src/server/Model/Cohort/DemographicAggregator.cs#L239) to "bucket" age >89 properly for use in the Visualization page.
- Rewrite the [CalculateAge](https://github.com/uwrit/leaf/blob/73d1ee7afc4f0b47193d0959f4e75e73964edba0/src/server/Services/Extensions/PatientDemographicExtensions.cs#L14) function in API.
- Update the [line of code in API](https://github.com/uwrit/leaf/blob/73d1ee7afc4f0b47193d0959f4e75e73964edba0/src/server/Model/Cohort/DatasetRecord.cs#L304) which specifically looks for patients older than 89 and reassigns a value.

Best,
-nic

**Comment by lawdank** [comment]
Hi @ndobb,

I wanted to pick this up again.
We would like to be able to implement this for compliance.  Please let me know if, by implementing your design suggestion I should raise a Pull Request. If that is possible, please let me know if, there is a checklist I should follow.


Thanks,
Sheersha

**Reply by ndobb** [reply]
Hi Sheersha,

There is no particular checklist to follow for implementing a pull request. If you and team make code changes, I will review them and run the code locally to confirm things work as expected, and if so, then I will merge them.

-nic
